src/d3d12/d3d12_resource_pool_texture.cpp

```diff
@@ -482,18 +480,18 @@ namespace wr
 		}
 		else
 		{
-			mip_lvls = metadata.mipLevels;
+			mip_lvls = static_cast<std::uint32_t>(metadata.mipLevels);
 		}
 
 		Format texture_format = static_cast<wr::Format>(metadata.format);
 
 		d3d12::desc::TextureDesc desc;
 
-		desc.m_width = metadata.width;
-		desc.m_height = metadata.height;
+		desc.m_width = static_cast<std::uint32_t>(metadata.width);
```

Comment:
All these static casts suppress the warning but a better solution would be to change the metadata.width type instead if possible.